### PR TITLE
Support company donations

### PR DESCRIPTION
**Pull request by lorentzlasson**
Still untested, and desired behavior is not fully nailed down

TODO
- [ ] Cover new functionality with tests
- [ ] Don't allow companies to be anonymous

### CODE REVIEW

File: src/@types/import/api/payment.types.ts [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
 
+export type BankPaymentRequest =
+  | (BaseRequest & BankPaymentDetails & IndividualPaymentDetails)
+  | (BaseRequest & BankPaymentDetails & CompanyPaymentDetails)
+
+export type SwishPaymentRequest =
+  | (BaseRequest & SwishPaymentDetails & IndividualPaymentDetails)
+  | (BaseRequest & SwishPaymentDetails & CompanyPaymentDetails)
+
+export type PaymentRequest = BankPaymentRequest | SwishPaymentRequest
```
Comment (lorentzlasson) [inline src/@types/import/api/payment.types.ts:42]: Switching to composition as inheritance gets tricky with multiple combinations

File: src/config/index.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+const getDomConfig = (attribute: string) => ROOT_NODE?.getAttribute(attribute)
+
+export const USE_DEV_DATA = process.env.REACT_APP_USE_DEV_DATA === 'true'
+export const API_URL = process.env.REACT_APP_API_URL
+
+export const CHOSEN_ORGANIZATION_ID =
+  getDomConfig('data-organization-id') ?? undefined
+
+// To support donations from companies instead of individuals
+export const IS_IN_COMPANY_MODE = getDomConfig('data-company-mode') === 'true'
```
Comment (lorentzlasson) [inline src/config/index.ts:12]: Made it configurable the same way "chosen organization" is - by adding an html attribute to the DOM.

File: src/content/texts.json [hunk 3]
```diff
@@ -41,6 +45,8 @@
 
       "anonymousDonor": {
         "name": "Anonym givare",
+        "companyName": "Anonym givare",
+        "organizationNumber": "",
```
Comment (lorentzlasson) [inline src/content/texts.json:49]: Not sure what would make sense to have here, or if companies should even be allowed to be anonymous

File: src/utils/validation.ts [hunk 4]
```diff
@@ -130,6 +111,28 @@ export const validateCreatePayment = (requestArgs: PaymentRequest) => {
     return false
   }
 
+  if (requestArgs._sourceType === 'individual') {
```
Comment (lorentzlasson) [inline src/utils/validation.ts:114]: Will add tests for `'company'` as soon as we've nailed down the desired behavior

File: src/components/panes/DonorPane/DonorPane.tsx [hunk 5]
```diff
@@ -112,6 +119,40 @@ export function DonorPane() {
...
+                  )}
+
+                  <TextInput
+                    type="text"
+                    placeholder={paneTexts.organizationNumberPlaceholder}
+                    {...register('organizationNumber', {
+                      required: !isAnonymous,
+                      validate: (val) =>
+                        val &&
+                        Validate.matches(val, /^(\d{1})(\d{5})\-(\d{4})$/),
```
Comment (lorentzlasson) [inline src/components/panes/DonorPane/DonorPane.tsx:145]: A regex I found on the web. Haven't tested extensively, but it lets my own org number through and it rejects some obviously wrong ones. Can add tests for it, if it's something we want to keep

File: src/config/index.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const ROOT_NODE = document.getElementById('root')
```
Comment (lorentzlasson) [inline src/config/index.ts:1]: I centralized all configuration I could find in this file. Just a preference of mine so if there are objections I'll revert

File: src/@types/import/api/payment.types.ts (hunk 1 above)
Comment (lnsndn) [reply src/@types/import/api/payment.types.ts:42]: Can you describe what will change about the payload?

File: src/components/panes/DonorPane/DonorPane.tsx (hunk 5 above)
Comment (lnsndn) [reply src/components/panes/DonorPane/DonorPane.tsx:145]: Seems to cover the most normal scenarios at least. Let's use it and see how it goes.

File: src/content/texts.json (hunk 3 above)
Comment (lnsndn) [reply src/content/texts.json:49]: Does seem like quite a strange state that perhaps we shouldn't allow

**Comment by lorentzlasson** [comment]
Not relevant anymore I don't think

**Pull request closed without merging**